### DISCUSSION TITLE
Query using OR from different tables throws an error

### DISCUSSION THREAD

**Discussion by juanantoniosantanamedina** [supabase/supabase#2234] [Questions]
Good morning all,

I might be missing something or maybe this is not allowed by Supabase.

I need to query some data from two tables which are related. I need to find one or more strings in some fields of both tables. This is what my query looks like

```javascript
	async testQuery() {
		const supabase = this.supabaseService.supabaseClient;

		const currentUSer = 'e32b4706-0213-40b9-bfc5-e325036f852f';
		const query = supabase
			.from<iUserNotification>('users_notifications')
			.select('*, sender:fk_id_users_profiles_sender(name_first)')
			.filter('fk_id_users_profiles', 'eq', currentUSer);

		query.or(`name_first.ilike.%juan%`, { foreignTable: 'sender' });
		query.or('title.ilike.%request%');

		const { data, error } = await query;
		if (error) {
			console.log('Error: ', error);
		} else {
			console.log('Data:', data);
		}
	}
```

That query does not return data, if I comment out one of the or sentences it works

Is it because I cannot use multiple tables in an OR? Am I not constructing the query properly?

Thanks in advance
Juan Antonio Santana

**Comment by steve-chavez** [comment]
**Edit: `or` is not applied across different foreign tables conditions now. See https://github.com/supabase/postgrest-js/issues/238**

Hey @juanantoniosantanamedina,

Try it like:

```js
supabase
	.from('users_notifications')
	.select('*, sender:fk_id_users_profiles_sender(name_first)')
	.or(`fk_id_users_profiles.eq.${currentUSer},title.ilike.%request%`);
	.or(`name_first.ilike.%juan%`, { foreignTable: 'sender' });
```

> Am I not constructing the query properly?

The conditions belonging to the same table need to be inside the [or](https://supabase.io/docs/reference/javascript/or#use-or-with-and) filter.

**Reply by juanantoniosantanamedina** [reply]
Thanks for your reply but that query does not filter the currentUser.
It returns the data even fk_id_users_profiles is not equal to currentUser. 

I guess it is because it is using an OR for it

This is what I'm trying to achieve:

```SQL
SELECT users_notifications.*, sender.name_first
FROM users_notifications INNER JOIN users_profiles AS sender
ON users_notifications.fk_id_users_profiles_sender=sender.id
WHERE users_notifications.fk_id_users_profiles='e32b4706-0213-40b9-bfc5-e325036f852f'
AND (
   users_notifications.title ILIKE '%request%'
   OR 
   sender.name_first ILIKE '%juan%'
)
```

**Reply by steve-chavez** [reply]
I see what you mean. You want to filter the source table(the one in FROM) rows based on the joined table conditions, this is a pending enhancement being tracked here https://github.com/supabase/postgrest-js/issues/197.

For now, a workaround would be wraping your query in a SQL function and call it on the js client through [rpc](https://supabase.io/docs/reference/javascript/rpc).

**Reply by rotimi-best** [reply]
> The conditions belonging to the same table need to be inside the or filter.

Thanks. I think this should be added to the docs. It wasn't clear to me that I only the queries on the same table should use the same `or` function.

**Reply by maulik-shneorweb** [reply]
any update enables us to write this query in supabase without rpc? 

 
> ```sql
> SELECT users_notifications.*, sender.name_first
> FROM users_notifications INNER JOIN users_profiles AS sender
> ON users_notifications.fk_id_users_profiles_sender=sender.id
> WHERE users_notifications.fk_id_users_profiles='e32b4706-0213-40b9-bfc5-e325036f852f'
> AND (
>    users_notifications.title ILIKE '%request%'
>    OR 
>    sender.name_first ILIKE '%juan%'
> )
> ```

**Reply by steve-chavez** [reply]
@maulik-shneorweb Not yet done but it's being worked on, you can track the progress on https://github.com/PostgREST/postgrest/issues/2563.

**Reply by maulik-shneorweb** [reply]
okay thanks for update @steve-chavez

**Reply by steve-chavez** [reply]
It's possible now to combine or conditions on different tables with https://postgrest.org/en/latest/references/api/resource_embedding.html#or-filtering-across-embedded-resources

**Comment by wilmerjpg** [comment]
Hi! Is there any solution for these cases already?
